src/modules/home/components/map/mapComponent.tsx

```diff
@@ -0,0 +1,55 @@
+import { mapContainerStyles } from './mapComponent.style';
+import { useState } from 'react';
+import { Map, Marker } from '@vis.gl/react-google-maps';
+import Box from '@mui/material/Box';
+
+interface Location {
+  lat: number;
+  lng: number;
+}
+
+interface MapClickEvent {
+  detail: {
+    placeId: string | null;
+    latLng: {
+      lat: number;
+      lng: number;
+    };
+  };
+}
+
+function CustomMap() {
+  const [selectedLocation, setSelectedLocation] = useState<Location | null>(null);
+  const [markerLocation, setMarkerLocation] = useState<Location>({
+    lat: 41.28332127984398,
+    lng: 69.2118501663208,
+  });
```

Comment:
What is the difference between selectedLocation and markerLocation? I see that you're giving them the same values below in the handleMapClick function.